tests/Type/Doctrine/data/QueryResult/expressionBuilderGetQueryNoObjectManager.php

```diff
@@ -0,0 +1,93 @@
+<?php declare(strict_types = 1);
+
+namespace QueryResult\CreateQuery;
+
+use Doctrine\ORM\AbstractQuery;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\QueryBuilder;
+use QueryResult\Entities\Many;
+use function PHPStan\Testing\assertType;
+
+class ExpressionBuilderGetQueryNoObjectManager
+{
+	public function isNullLiteralString(EntityManagerInterface $em): void
+	{
+		$result = $em->createQueryBuilder()->expr()->isNull('field');
+		assertType('literal-string&non-empty-string', $result);
+	}
+
+	public function isNullNonLiteralString(EntityManagerInterface $em): void
+	{
+		$field = strtolower('field'); // Non literal-string, e.g. $_POST['field'];
```

Comment:
I don't think it's a good idea to use `strtolower` in the test.

You should instead do
```
public function isNullNonLiteralString(EntityManagerInterface $em, string $field): void
```

Because strtolower of a literal string is a literal string. One day we'll implement the conditional type or dynamic return type extension in phpstan and it will force to update all the tests here.

Reply:
I'm happy to change this, but I don't think it needs it... the `literal-string` type is following the [is_literal() RFC](https://wiki.php.net/rfc/is_literal) (which I really need to do a second version of), and the only modification that can be done to a `literal-string` is when it's concatenated with another `literal-string` (to help adoption). The only native functions that can return a `literal-string` are `str_repeat()`, `str_pad()`, `implode()`, and `join()`; as these functions simply concatenate their inputs. I did mention in the RFC about supporting other functions, and noted that `strtoupper()` might be reasonable, but it was decided to not allow any other string modifications, to keep the implementation simpler, not affect performance, and make it easier to prove the security.